### ISSUE TITLE
access to this.$options

### ISSUE THREAD

**Issue by joseporto** [vue-styleguidist/vue-styleguidist#485]
Maybe I'm missing something, but I'm having trouble accessing `this.$options` object from my components.

This is how they are declared:

```js
// src/components/index.js

import ComponentA from './Components/ComponentA'
import ComponentB from './Components/ComponentB'

import defaults from './helpers/defaults'

const Components = {
  install (Vue, options) {
    const config = Object.assign(defaults, options)

    Vue.component(ComponentA.name, { extends: ComponentA, config })
    Vue.component(ComponentB.name, { extends: ComponentB, config })
  }
}

if (typeof window !== 'undefined' && window.Vue) {
  window.Vue.use(Components);
}

export default Components
export {ComponentA, ComponentB}
```

This is how my app imports them:

```js
// src/main.js

import Vue from 'vue'
import App from './App.vue'
import components from './components/index.js'

Vue.use(components, {
    optionA: 'test',
    optionB: false
  }
})

new Vue({
  render: h => h(App),
}).$mount('#app')
```

And when running my app, `this.$options` is available, but in styleguideist they are not and I get this error:

> [Vue warn]: Error in render: "TypeError: undefined is not an object (evaluating 'this.$options.config.optionA')"

-------

Before submitting a new question, make sure you:

- [x] Searched existing [Stack Overflow questions](https://stackoverflow.com/questions/tagged/vue-styleguidist).
- [x] Searched opened and closed [GitHub issues](https://github.com/vue-styleguidist/vue-styleguidist/issues?utf8=%E2%9C%93&q=is%3Aissue).
- [x] Read [the documentation](https://vue-styleguidist.github.io/getting-started.html).
- [x] Googled your question.

**Comment by elevatebart** [comment]
Hello @joseporto 

As you probably know styleguidist only looks at your components file. It, therefore, has no idea of the options you passed to Vue in `main.js`. You will have to help it a little.

I would suggest configuring the [renderRootJsx](https://vue-styleguidist.github.io/Configuration.html#renderrootjsx) option.

It expects a file. In your case, the file you would have would look like this:

```js
import Vue from "vue";

Vue.use(components, {
    optionA: 'test',
    optionB: false
  }
})

export default previewComponent => {
  // https://vuejs.org/v2/guide/render-function.html
  return {
    render(createElement) {
      return createElement(previewComponent);
    }
  };
};
```

I hope this helps,
Bart

**Comment by joseporto** [comment]
Thanks @elevatebart. It does make sense, but still not working. I tried to find projects that use the same pattern, but couldn’t find one so far.

**Comment by elevatebart** [comment]
I might point you toward the [nuxtjs example](https://github.com/vue-styleguidist/vue-styleguidist/tree/dev/examples/nuxtjs) that uses a similar pattern for faking the use of the nuxt axios plugin.
See inside of `examples/nuxtjs/styleguide/styleguide.root.js`

**Comment by elevatebart** [comment]
Did it help?
Can you share what you put in $options?

Come on over to the discord channel on vue-land, I am sure someone can help ;)

**Comment by joseporto** [comment]
Unfortunately not. Although I did follow the example, and it totally made sense to change the prototype it self. Eventually I ended up going for a `sass` solution to override variables before importing the component styles, since this actual package would benefit from being configured and styled via CSS and not javascript. But I have another component in the making which, due to it's design and functionality, will benefit from this type of initialisation. So, I won't close the issue just yet, since I haven't myself provided a solution. But feel free to close it, and I can always come back to it once I'm able to do invest some time on my new component.
